### PR TITLE
Asking critters to identify themselves with vog works

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Asking basic or simple mobs to identify themselves will now actually get them to say their REAL name

image

## Why It's Good For The Game

~IDENTIFY
the kid named cockroach:~

I like this as a small easter egg functionality, being able to defacto name critters stuff.

## Changelog

:cl:
add: You can now ask critters their name with voice of god. Their ACTUAL name.
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
No more Cockroach spam........ 😔
image

**Comment by Shadowflame909** [comment]
we love more cross-feature interaction

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/voice_of_god_command.dm [hunk 1]
```diff
@@ -253,15 +249,32 @@ GLOBAL_LIST_INIT(voice_of_god_commands, init_voice_of_god_commands())
...
 	for(var/mob/living/target as anything in listeners)
-		addtimer(CALLBACK(target, /atom/movable/proc/say, target.real_name), 0.5 SECONDS * iteration)
+		addtimer(CALLBACK(src, .proc/state_name, target), 0.5 SECONDS * iteration)
 		iteration++
 
+///just states the target's name, but also includes the renaming funny.
+/datum/voice_of_god_command/who_are_you/proc/state_name(mob/living/target)
+	if(QDELETED(target))
+	   return
+	if(isanimal_or_basicmob(target) && target.name == initial(target.name) && target:gold_core_spawnable == FRIENDLY_SPAWN)
```
Comment (Watermelon914) [inline code/datums/voice_of_god_command.dm:266]: you should just properly typecast here instead of using :

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/datums/voice_of_god_command.dm [hunk 2]
```diff
@@ -253,15 +249,39 @@ GLOBAL_LIST_INIT(voice_of_god_commands, init_voice_of_god_commands())
...
 
+///just states the target's name, but also includes the renaming funny.
+/datum/voice_of_god_command/who_are_you/proc/state_name(mob/living/target)
+	if(QDELETED(target))
+		return
+	var/gold_core_spawnable = NO_SPAWN
+	if(isbasicmob(target))
+		var/mob/living/basic/basic_bandy = target
+		gold_core_spawnable = basic_bandy.gold_core_spawnable
+	if(isanimal(target))
```
Comment (ShizCalev) [inline code/datums/voice_of_god_command.dm:270]:
```suggestion
	else if(isanimal(target))
```

**Review by ShizCalev — APPROVED** [review]

**Comment by LePiromano** [comment]
https://user-images.githubusercontent.com/55637865/189505091-081b4f26-6cc1-4f80-aee2-0d2ad95ef6f0.mp4

**Pull request merged**